AttitudeManager/ControlAlgorithms/Src/AM_FixedControl.cpp

```diff
@@ -1,17 +1,116 @@
 #include "AM_FixedControl.hpp"
+#include "PID.hpp"
+
+ // #include math_constants to remove sloppy definitions
+constexpr double ZP_PI = 3.14159265358979311599796346854; // more precise pi value
+
+static float DEG_TO_RAD(float angleInDegrees); 
+
+static float DEG_TO_RAD(float angleInDegrees) {
+    return ((angleInDegrees) * ZP_PI / 180.0);
+}
+// Remove in future to mitigate cluttering
+typedef struct {
+    float roll, pitch, yaw;             // Degrees. Yaw of 180 is north.
+    float rollRate, pitchRate, yawRate; // Degrees/second
+    float airspeed;                     // m/s
+    float altitude;                     // m
+    float rateOfClimb;                  // m/s
+    long double latitude;               // Decimal degrees
+    float latitudeSpeed;                // m/s
+    long double longitude;              // Decimal degrees
+    float longitudeSpeed;               // m/s
+    double track;                       // Degrees. Track of 0 is north.
+    float groundSpeed;                  // m/s
+    double heading;                     // Degrees. Heading of 0 is north.
+} SFOutput_t;
 
 namespace AM {
 
 void FixedControl::runControlsAlgo(const AttitudeManagerInput &instructions,
                                    float outputs[],
                                    uint8_t outputsLength) const {
+   
+   // Get current attitude from sensorfusion 
+    SFOutput_t currentAttitude; // TODO: This needs to be retrieved from LOS
+    
+    // Compute target values
+    float targetHeading = currentAttitude.heading - instructions.heading;
+    float targetAltitude = instructions.z_dir; 
+    float targetPitch = instructions.x_dir * maxPitchAngle;
+    float targetBank = instructions.y_dir * maxBankAngle;
+     
+    // Adjust heading difference for the PID
+    if (targetHeading <= (-180)) {
+        targetHeading += 180;
+    } else if (targetHeading > 180){
+        targetHeading -=360;
+    }
+   
+    // Define PID values for mixing
+    PIDController pid_bank{bank_kp,        bank_ki,      bank_kd, 
+                           bank_i_windup, -maxBankAngle, maxBankAngle };
+    PIDController pid_rudder{rudder_kp,       rudder_ki, rudder_kd, 
+                             rudder_i_windup, -100,      100};
+    PIDController pid_pitch{pitch_kp,       pitch_ki,       pitch_kd, 
+                            pitch_i_windup, -maxPitchAngle, maxPitchAngle};
+    PIDController pid_airspeed{airspeed_kp,       airspeed_ki,  airspeed_kd, 
+                               airspeed_i_windup, airspeed_min, airspeed_max};                        
+
+    // Compute values for mixing
+    float bank = pid_bank.execute(targetBank, currentAttitude.roll, currentAttitude.rollRate);
+
+    float pitch = pid_pitch.execute(targetPitch, currentAttitude.pitch, currentAttitude.pitchRate);
+
+    float rudderSet = rudderPercent(bank);
```

Comment:
I'm assuming you're referring to "rudderSet." I did it this way because I wasn't sure what to do with the rudder. Referencing a comment above, if we have multiple rudders, would each need its own pid.execute() thing? Maybe the rudder is done like the other PID values (i.e., rudder = pid_rudder.execute();)?